vp/workflow/views.py

```diff
@@ -103,27 +105,72 @@ def save_workflow(request):
         'message': 'Saved the graph to ' + workflow.file_path + '!'
     })
 
-def retrieve_nodes_for_user(request):
-    if request.method =='GET':
+def retrieve_nodes_for_ide(request):
+    if request.method == 'GET':
+        """Retrieve all nodes that a user can have access to in the IDE from a specified file in request body.
+        
+        Args:
+            request: Django request Object
+       
+         Returns:
+            200 - JSON response with data.
+            400 - No file specified/issues reading json file.
+            404 - File specified not found, or not JSON graph.
         """
-        Retrieve all nodes that a user can have access to in the IDE.
-        Currently returning default set of nodes. 
-        //TODO pick these node files from a file in the system.
-        """
-        data = {
-            "I/O": [
-                {"key": "read-csv", "name": "Read CSV", "numPortsIn": 0, "color": "black"}
-            ],
-            "Manipulation": [
-                {"key": "filter", "name": "Filter Rows", "color": "red"},
-                {"key": "pivot", "name": "Pivot Table", "color": "blue"},
-                {"key": "multi-in", "name": "Multi-Input Example", "numPortsIn": 3, "color": "green"}
-            ]
-        }
+        try:
+            request_body = json.loads(request.body)
```

Comment:
Is there a benefit to this approach vs. `request.GET.get('file')`? Django [stores all GET parameters in an dictionary](https://docs.djangoproject.com/en/3.0/ref/request-response/#django.http.HttpRequest.GET) already, so I don't think you need to parse the body.

Reply:
Please see explanation above.

Reply:
Is this intended to load a server-side JSON file by name, or accept an upload from the client? [The body of a GET request should be ignored](https://stackoverflow.com/questions/978061/http-get-with-request-body), so if it's the former, it needs to go in the querystring, and if it's the latter it needs to be a POST.